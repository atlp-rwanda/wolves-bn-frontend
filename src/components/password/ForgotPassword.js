import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Link } from 'react-router-dom';
import { findUser } from '../../redux/actions/forgot.password';
import InputField from '../common/InputField';
import Button from '../common/Button';
import './password.scss';

const ForgotPassword = () => {
  const newState = useSelector(state => state);
  const dispatch = useDispatch();

  const [email, setEmail] = useState('');

  const [submitted, setSubmitted] = useState(false);

  return (
    <div className='container'>
      <div className='form-pass'>
        <form className='resetPassword' onSubmit={(e) => {
          e.preventDefault();
          console.log(email);
          setSubmitted(true);

          if (email) {
            dispatch(findUser(email));
          }
        }}>
          {newState.findUser.message === 'Check your email for the reset link' ? <div className= 'valid-feedback'>{newState.findUser.message}</div>
            : <div className='invalid-feedback'>{newState.findUser.message} <Link to='/signup' ><h4 className='linkTo'><span>Create Account</span> </h4></Link>
            </div> }
          <div className='form-group'>
            <h3>Forgot Password</h3>
          </div>
          <div className='form-group'>
          <InputField
           type='email'
           name='email'
           label='Email'
           className='form-control'
           onChange = {(e) => { setEmail(e.target.value); }}
          />
          {submitted && !email && <div className="invalid-feedback">Email is required</div>}
          </div>
          <div className='form-group'>
          <Button content='Submit'
          />
          </div>
        </form>
        </div>
    </div>
  );
};

export default ForgotPassword;