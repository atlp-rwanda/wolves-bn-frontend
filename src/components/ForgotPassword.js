import React, { Component } from 'react';
import { connect } from 'react-redux';
import { findUser } from '../redux/actions/forgot.password';
import InputField from './common/InputField';
import Button from './common/Button';

class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      message: ''
    };

    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
  }

  onChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  onSubmit(e) {
    e.preventDefault();
    const { email } = this.state;
    const { findingUser } = this.props;

    findingUser(email)
      .then(() => {
        console.log('Ngaaho');
      })
      .catch(() => {
        console.log('noooo');
      });
  }

  render() {
    const { message } = this.props;
    console.log(message);
    return (
      <div>
        {message && (
              <div className="form-group">
                <div className="alert alert-danger" >
                  {message}
                </div>
              </div>
        )}

        <form onSubmit={this.onSubmit}>
          <InputField
           type='email'
           name='email'
           label='Email'
           className='form-control'
           handleChange= {this.onChange}
          />
          <Button content='Submit'
          />
        </form>
      </div>
    );
  }
}

function mapDispatchToProps(dispatch) {
  return {
    findingUser: (email) => dispatch(findUser(email))
  };
}

function mapStateToProps(state) {
  const { message } = state.reset;
  return {
    message
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPassword);