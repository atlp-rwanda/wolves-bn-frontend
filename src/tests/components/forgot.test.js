import React from 'react';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import ForgotPassword from '../../components/ForgotPassword';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('forgot password component', () => {
  const storeBfr = mockStore({
    reset: {
      message: ''
    },
  });

  storeBfr.dispatch = jest.fn();

  const component = renderer.create(
    <Router>
    <Provider store={storeBfr}>
      <ForgotPassword />
    </Provider>
    </Router>
  );
  it('should render with initial state', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});