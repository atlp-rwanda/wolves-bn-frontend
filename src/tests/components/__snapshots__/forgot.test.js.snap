// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forgot password component should render with initial state 1`] = `
<div>
  
  <form
    onSubmit={[Function]}
  >
    <div
      className="inputDiv"
    >
      <label>
        Email
      </label>
      <input
        name="email"
        type="email"
      />
    </div>
    <button
      className="btn"
    >
      Submit
    </button>
  </form>
</div>
`;
